### DISCUSSION TITLE
Is R513A supported in CoolProp fluids list

### DISCUSSION THREAD

**Discussion by yolandaLiuZongPing** [CoolProp/CoolProp#2282] [Q&A] [answered]
I've tried to print out the list of CoolProp Fluids, but R513A is not included, how can I do if I need R513A? here is the list of my version: 1-Butene,Acetone,Air,Ammonia,Argon,Benzene,CarbonDioxide,CarbonMonoxide,CarbonylSulfide,cis-2-Butene,CycloHexane,Cyclopentane,CycloPropane,D4,D5,D6,Deuterium,Dichloroethane,DiethylEther,DimethylCarbonate,DimethylEther,Ethane,Ethanol,EthylBenzene,Ethylene,EthyleneOxide,Fluorine,HeavyWater,Helium,HFE143m,Hydrogen,HydrogenChloride,HydrogenSulfide,IsoButane,IsoButene,Isohexane,Isopentane,Krypton,m-Xylene,MD2M,MD3M,MD4M,MDM,Methane,Methanol,MethylLinoleate,MethylLinolenate,MethylOleate,MethylPalmitate,MethylStearate,MM,n-Butane,n-Decane,n-Dodecane,n-Heptane,n-Hexane,n-Nonane,n-Octane,n-Pentane,n-Propane,n-Undecane,Neon,Neopentane,Nitrogen,NitrousOxide,Novec649,o-Xylene,OrthoDeuterium,OrthoHydrogen,Oxygen,p-Xylene,ParaDeuterium,ParaHydrogen,Propylene,Propyne,R11,R113,R114,R115,R116,R12,R123,R1233zd(E),R1234yf,R1234ze(E),R1234ze(Z),R124,R1243zf,R125,R13,R134a,R13I1,R14,R141b,R142b,R143a,R152A,R161,R21,R218,R22,R227EA,R23,R236EA,R236FA,R245ca,R245fa,R32,R365MFC,R40,R404A,R407C,R41,R410A,R507A,RC318,SES36,SulfurDioxide,SulfurHexafluoride,Toluene,trans-2-Butene,Water,Xenon

**Comment by yolandaLiuZongPing** [comment]
I was wondering can I do something like this: CP.PropsSI('D','T',300,'P',101325,'REFPROP::R1234yf[0.56]&R134a[0.44]') to get hte R513A.  can I do a similar operation to get the list of these mixtures :
![image](https://github.com/CoolProp/CoolProp/assets/24609311/b63e5a90-a05c-458c-8372-e9e32a37a06e)

**Comment by ibell** [comment, marked as answer]
Yes, you can do something like that, but:
1) You should use the low-level interface
2) You need to provide molar fractions

More generally it would be useful for someone to add all the predefined mixtures into CoolProp.

**Comment by yolandaLiuZongPing** [comment]
Thank you very much, Ian. I have remarked this conversation as concluded. To elaborate, R513A comprises 44% R134a and 56% 1234yf in terms of molar composition. 
I'm relatively new to the field of thermodynamics. 
In the coming times, I plan to make contributions to CoolProp by incorporating all the pre-defined mixture data if I'll have some spare time.

**Reply by HightopRaven** [reply]
for clarity, if anyone is reading this in posterity, R513A composition is 44.0% R134a and 56.0% R1234yf by mass fraction, not by molar fraction
https://www.ashrae.org/technical-resources/standards-and-guidelines/ashrae-refrigerant-designations
https://www.ashrae.org/technical-resources/standards-and-guidelines/read-only-versions-of-ashrae-standards
https://ashrae.iwrapper.com/ASHRAE_PREVIEW_ONLY_STANDARDS/STD_34_2022

**Comment by Ed-Tronics** [comment]
I also am in need of R513A, I've tried the solution above but it throws an exception:

`CoolProp Exception: key [R134A[0.440]&R1234yf[0.560]] was not found in string_to_index_map in JSONFluidLibrary` 
I'm currently reading the source code to figure out why this might be. Seems that it recognises I want to do a cubic Helmholtz energy transformation but doesn't understand my string.

Our project currently uses CoolProp version 6.1.0 - could this be the issue? Would there be many changes required if I update to the latest version?

**Comment by ibell** [comment]
For R513A, interaction parameters are needed for R-134a/1234yf. They are
available in http://www.coolprop.org/zbibliography.html#id32, so I think
your calculation (at least the model instantiation) should work. To be
clear, the string definition of composition is MOLAR composition.  In the
low-level interface you can specify mass compositions, but not in the
high-level one.

**Comment by pcjc2** [comment]
Is it possible to access these non pre-defined mixtures from the Excel interface?  (I'm doing basic modelling in Excel, as it is quite handy for building graphs).

**Reply by ibell** [reply]
Yes, there are methods, the helper function for constructing the string is called MixtureString. Example: 

![image](https://github.com/user-attachments/assets/9ae74203-f161-469a-8652-b399d6828865)

**Comment by lightsrout** [comment]
@Ed-Tronics: did you end up getting it working?
@ibell: I just reinstalled coolprop version 8.0.0 and can not access R513a with the high- or low-level interface (See screenshot).
<img width="1373" height="861" alt="CoolProp R513a" src="https://github.com/user-attachments/assets/7ed896f0-2798-4ddb-af62-885fbd27bde7" />
Any idea what might be the issue?

**Reply by ibell** [reply]
Yes, case-sensitivity. "R513A", not "R513a"

**Reply by ibell** [reply]
```
mkdir uvv8
nano main.py # (edit as below)
uv run --with coolprop==8.0.0 python main.py
uv run --with coolprop==7.2.0 python main.py
```
with `main.py` of 
```python

import CoolProp
CP = CoolProp.CoolProp

def main():
    print(CoolProp.__version__)
    AS = CP.AbstractState('HEOS','R513A.mix')
    print(AS.fluid_names())
    AS.update(CP.PT_INPUTS, 101325, 300)
    print(AS.rhomass(), "kg/m^3")

if __name__ == "__main__":
    main()

```
yielding
```
(base) ianbell@Ians-MacBook-Air-2 uvv8 % uv run --with coolprop==8.0.0 python main.py
8.0.0
['R1234yf', 'R134a']
4.4961339199741195 kg/m^3
(base) ianbell@Ians-MacBook-Air-2 uvv8 % uv run --with coolprop==7.2.0 python main.py
7.2.0
['R1234yf', 'R134a']
4.496847614687022 kg/m^3
```

so you can see that in both versions you want `"R513A.mix"` and that the mixture model parameters (and the pure fluid EOS for R1234yf) have slightly changed

**Reply by lightsrout** [reply]
Right, that was it. It works now for me too now. Thank you so much!!
